### PR TITLE
Implement new AppendMissingKey method on error struct

### LINKED ISSUE #56: Create new method on Error struct to handle appending of missingkeys
In a couple places in the setting code base the below functionality can be seen:

```go

		errs.missingKeys = append(errs.missingKeys, key)
```

It's been suggested (@alistairjudson) that replacing this with a more elegant solution could be accomplished by implementing a method associated with the ProviderError struct to allow for this functionality to be held closer to the struct itself. An example of the type of functionality can be seen below:

```go
type Error struct{
	missingKeys []string
}

func(e *Error) Append(key string) *Error {
	if e == nil {
		return &Error{
			missingKeys: []string{key},
		}
		
	}
	e.missingKeys = append(e.missingKeys, key)
	return e
} 
```

This issue is to implement that functionality.

### PR DESCRIPTION
**Pull request by GitCodeOli**
A new append method is added to the ProviderError struct for better handling for adding missingkeys to provider errors.

ProviderError methods now all use pointers 

Closes #56

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -32,24 +32,33 @@ type ProviderError struct {
 	missingKeys []string
 }
 
+// AppendMissingKey either adds a key to a pre-existing slice of keys or instantiates a new slice of keys
+func (pe *ProviderError) AppendMissingKey(key string) *ProviderError {
+	if pe == nil {
+		return NewProviderError(key)
```
Comment (alessandrozucca) [inline error.go]: thinking about it, we should be able to do 
```
if pe == nil {
  pe = &ProviderError{}
}
```

https://play.golang.org/p/tbUBsYTye9V

Comment (alessandrozucca) [reply error.go]: and change the method signature to `func (pe *ProviderError) AppendMissingKey(key string)` so then we change it to

`pe.AppendMissingKey(key)`

Comment (alessandrozucca) [reply error.go]: I take it back.
Although the assignment works, as soon as we get out of the scope of the function the allocation disappears.

File: error.go [hunk 2]
```diff
@@ -32,24 +32,33 @@ type ProviderError struct {
 	missingKeys []string
 }
 
+// AppendMissingKey either adds a key to a pre-existing slice of keys or instantiates a new slice of keys
+func (pe *ProviderError) AppendMissingKey(key string) *ProviderError {
```
Comment (geototti21) [inline error.go]: Suggestion, why dont we change this func to `func (pe *ProviderError) AppendMissingKeys(keys ...string) *ProviderError ` so we can append more than one keys in once.

Comment (GitCodeOli) [reply error.go]: ^Sure will do

**Pull request merged**